### PR TITLE
Add continuous integration for gitlab-ci

### PR DESCRIPTION
**Pull request by yanicasa**
new CI for gitlab is massively generic to allow each group to be able to use it directly without modifications in the CI source code regardless of their runtime environment. 
Adaptations to the environment are done only in the settings of the projects. (paths of tools, version of tools, urls of repositories to use...)

### CODE REVIEW

File: .gitlab-ci/cv32.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (MikeOpenHWGroup) [inline .gitlab-ci/cv32.yml:1]: This file will not be necessary as the original version of it (`.gitlab-ci.yml`) is no longer in use.

**Review by MikeOpenHWGroup — APPROVED** [review]
This all looks good @yanicasa.  The `cv32.yml` will not be necessary (but thanks for considering it!).

File: .gitlab-ci/cv32.yml (hunk 1 above)
Comment (yanicasa) [reply .gitlab-ci/cv32.yml:1]: would you prefer I remove it or leave it as an entry point to motivate someone to help set up a cv32 regression?

Comment (MikeOpenHWGroup) [reply .gitlab-ci/cv32.yml:1]: Originally I was thinking that it should be removed.  Your idea to leave it as an entry point for future cv32 regressions on from GitLab is a good one.  Perhaps add a comment to indicate that it is currently not used.  Thanks!

**Review by JeanRochCoulon — APPROVED** [review]

**Pull request merged**